wkcuber/api/layer.py

```diff
@@ -97,38 +98,43 @@ def get_mag(self, mag: Union[int, str, list, tuple, np.ndarray, Mag]) -> MagView
 
         This function raises an `IndexError` if the specified `mag` does not exist.
         """
-        mag = Mag(mag).to_layer_name()
+        mag = Mag(mag)
         if mag not in self.mags.keys():
-            raise IndexError("The mag {} is not a mag of this layer".format(mag))
+            raise IndexError(
+                "The mag {} is not a mag of this layer".format(mag.to_layer_name())
+            )
         return self.mags[mag]
 
     def add_mag(
         self,
         mag: Union[int, str, list, tuple, np.ndarray, Mag],
         block_len: int = 32,
         file_len: int = DEFAULT_WKW_FILE_LEN,
-        block_type: int = wkw.Header.BLOCK_TYPE_RAW,
+        compress: bool = False,
     ) -> MagView:
         """
         Creates a new mag called and adds it to the layer.
-        The parameter `block_len`, `file_len` and `block_type` can be
+        The parameter `block_len`, `file_len` and `compress` can be
         specified to adjust how the data is stored on disk.
 
         The return type is `wkcuber.api.mag_view.MagView`.
 
         Raises an IndexError if the specified `mag` already exists.
         """
         # normalize the name of the mag
-        mag = Mag(mag).to_layer_name()
+        mag = Mag(mag)
+        block_type = (
+            wkw.Header.BLOCK_TYPE_LZ4HC if compress else wkw.Header.BLOCK_TYPE_RAW
+        )
```

Comment:
I was using `LZ4HC` somewhere else in the API. Therefore, I decided to also use `LZ4HC` here. Do you think 'LZ4' would be better?

Reply:
`LZ4HC` seems fine as a default for now. This improves the compression ratio and slows down the writing therefore (compared to `LZ4`), but reading does not seem to get worse, according to [this SO post](https://stackoverflow.com/questions/28635496/difference-lz77-vs-lz4-vs-lz4hc-compression-algorithms).
We're using `LZ4` so far in vx, but that was never conscious decision against the HC version.